### PR TITLE
Removed deprecated code no longer needed after migrating to resource contracts

### PR DESCRIPTION
**Pull request by kopper**
See commit by commit...

### CODE REVIEW

File: pkg/orchestration/wiring/ec2compute/common/common.go [hunk 1]
```diff
@@ -138,42 +138,13 @@ func WireUp(microServiceNameInSpec, ec2ComputePlanName string, stateResource *or
...
-			continue
-		}
-		// DEPRECATED: binding using raw SmithResources, use resource contracts instead
-		for _, resource := range dependency.SmithResources {
-			// TODO can be plugin
-			if resource.Spec.Plugin != nil {
-				continue
-			}
-			if si, ok := resource.Spec.Object.(*sc_v1b1.ServiceInstance); ok {
-				// depend directly on compute
```
Comment (kopper) [inline pkg/orchestration/wiring/ec2compute/common/common.go]: ⚠️ we are loosing ability for one EC2Compute to depend on another one

File: pkg/orchestration/wiring/sqs/plugin.go [hunk 2]
```diff
@@ -78,6 +78,11 @@ func WireUp(stateResource *orch_v1.StateResource, context *wiringplugin.WiringCo
 	wiredResources = append(wiredResources, serviceInstance)
 
 	result := &wiringplugin.WiringResult{
+		Contract: wiringplugin.ResourceContract{
```
Comment (kopper) [inline pkg/orchestration/wiring/sqs/plugin.go]: this contract was missing

File: pkg/orchestration/wiring/ec2compute/common/common.go [hunk 3]
```diff
@@ -138,42 +138,13 @@ func WireUp(microServiceNameInSpec, ec2ComputePlanName string, stateResource *or
 	var references []smith_v1.Reference
 
 	for _, dependency := range dependencies {
-		bound := false
-		if bindableShape, ok := dependency.Contract.FindShape(knownshapes.BindableEnvironmentVariablesShape); ok {
-			resourceReference := bindableShape.(*knownshapes.BindableEnvironmentVariables).Data.ServiceInstanceName
-			bindingResources = append(bindingResources, wiringutil.ConsumerProducerServiceBindingV2(stateResource.Name, dependency.Name, resourceReference, false))
-			continue
-		}
-		// DEPRECATED: binding using raw SmithResources, use resource contracts instead
```
Comment (kopper) [inline pkg/orchestration/wiring/ec2compute/common/common.go]: 💥

File: pkg/orchestration/wiring/k8scompute/plugin.go [hunk 4]
```diff
@@ -132,38 +132,15 @@ func WireUp(resource *orch_v1.StateResource, context *wiringplugin.WiringContext
...
-			binding := wiringutil.ConsumerProducerServiceBindingV2(resource.Name, dep.Name, resourceReference, false)
-			smithResources = append(smithResources, binding)
-			bindingResources = append(bindingResources, binding)
-			continue
+		bindableShape, found := dep.Contract.FindShape(knownshapes.BindableEnvironmentVariablesShape)
+		if !found {
+			return nil, false, errors.Errorf("cannot depend on resource %q of type %q, only dependencies providing shape %q are supported", dep.Name, dep.Type, knownshapes.BindableEnvironmentVariablesShape)
 		}
 
-		// DEPRECATED: binding using raw SmithResources, use resource contracts instead
```
Comment (kopper) [inline pkg/orchestration/wiring/k8scompute/plugin.go]: 💥

File: pkg/orchestration/wiring/wiringplugin/types.go [hunk 5]
```diff
@@ -46,8 +46,6 @@ type WiredDependency struct {
 	Name     voyager.ResourceName
 	Type     voyager.ResourceType
 	Contract ResourceContract
-	// DEPRECATED: use Contract
-	SmithResources []smith_v1.Resource
```
Comment (kopper) [inline pkg/orchestration/wiring/wiringplugin/types.go]: 💥

File: pkg/orchestration/wiring/wiringplugin/types.go [hunk 6]
```diff
@@ -159,8 +159,6 @@ type WiringResult struct {
 
 type WiredSmithResource struct {
 	SmithResource smith_v1.Resource
-	// DEPRECATED: use Shapes, Refs and/or Data in ResourceContract to expose information
-	Exposed bool
```
Comment (kopper) [inline pkg/orchestration/wiring/wiringplugin/types.go]: 💥

File: pkg/orchestration/wiring/wiringutil/iam/iam.go [hunk 7]
```diff
@@ -58,12 +58,11 @@ func PluginServiceInstance(computeType iam_plugin.ComputeType, resourceName voya
...
 			},
 		},
-		Exposed: false, // Why anything would want to depend on ServiceInstance that provisions the IAM role?
 	}
 
 	return instanceResource, nil
 }
 
 func ServiceBinding(compute voyager.ResourceName, iamPluginServiceInstance smith_v1.ResourceName) wiringplugin.WiredSmithResource {
-	return wiringutil.ResourceInternalServiceBinding(compute, iamPluginServiceInstance, namePostfix, true)
```
Comment (kopper) [inline pkg/orchestration/wiring/wiringutil/iam/iam.go]: @ash2k can this be a problem? we no longer expose this binding

File: pkg/orchestration/wiring/wiringplugin/types.go [hunk 8]
```diff
@@ -154,11 +154,7 @@ func (c *ResourceContract) FindShape(shapeName ShapeName) (Shape, bool /* found
 
 type WiringResult struct {
 	Contract  ResourceContract
-	Resources []WiredSmithResource
-}
-
-type WiredSmithResource struct {
```
Comment (kopper) [inline pkg/orchestration/wiring/wiringplugin/types.go]: 💥

File: pkg/orchestration/wiring/wiringutil/service_binding.go [hunk 9]
```diff
@@ -16,20 +16,7 @@ import (
 // there is a need to bind to multiple ServiceInstances produced by the same producer, construct names
 // (using ConsumerProducerResourceNameWithPostfix() and ConsumerProducerMetaNameWithPostfix() functions) with
 // different postfixes.
-// DEPRECATED: use ConsumerProducerServiceBindingV2 instead
```
Comment (kopper) [inline pkg/orchestration/wiring/wiringutil/service_binding.go]: 💥

File: pkg/orchestration/wiring/wiringutil/iam/iam.go (hunk 7 above)
Comment (ash2k) [reply pkg/orchestration/wiring/wiringutil/iam/iam.go]: What do you mean? We no longer provide a way for consumer to depend on something directly.

**Comment by ash2k** [comment]
I'm merging this so that I can rebase my branch and merge it too =) Don't want to merge my one first to avoid creating work for you. Thanks for all the refactoring! Awesomely done!

**Pull request merged**